src/toil/server/wes/toil_backend.py

```diff
@@ -0,0 +1,335 @@
+        return os.path.isdir(self.work_dir)
+
+    def get_state(self) -> str:
+        """ Return the state of the current run."""
+        return self.fetch("state", "UNKNOWN")
+
+    def set_state(self, state: str) -> None:
+        """ Set the state for the current run."""
+        with open(os.path.join(self.work_dir, "state"), "w") as f:
+            f.write(state)
+
+    def set_up_run(self) -> None:
+        """ Set up necessary directories for the run."""
+        if not os.path.exists(self.exec_dir):
+            os.makedirs(self.exec_dir)
+
+    def clean_up(self) -> None:
+        """ Clean directory and files related to the run."""
+        shutil.rmtree(os.path.join(self.work_dir))
+
+    def get_run_command(self, request: Dict[str, Any], options: List[str]) -> List[str]:
+        """
+        Return a list of commands, and the working directory for the run.
+        """
+        with open(os.path.join(self.work_dir, "request.json"), "w") as f:
+            json.dump(request, f)
+
+        options = [f"--opt={option}" for option in options]
+        return [resolveEntryPoint("_toil_wes_runner"), "--work_dir", self.work_dir, *options]
+
+    def get_output_files(self) -> Any:
+        """
+        Return a collection of output files that this workflow generated.
+        """
+        return json.loads(self.fetch("outputs.json", "{}"))
+
+
+class ToilBackend(WESBackend):
+    """
+    WES backend implemented for Toil to run CWL, WDL, or Toil workflows. This
+    class is responsible for validating and executing submitted workflows.
+
+    Local implementation -
+    Interact with the "workflows/" directory in the filesystem to store and
+    retrieve data associated with the runs.
+    """
+
+    def __init__(self, work_dir: str, options: List[str]) -> None:
+        super(ToilBackend, self).__init__(options)
+        self.work_dir = work_dir
+        self.supported_versions: Dict[str, List[str]] = {}
+        self.processes: Dict[str, "subprocess.Popen[bytes]"] = {}
+
+    def register_wf_type(self, name: str, supported_versions: List[str]) -> None:
+        """
+        Register a workflow type that this backend should support.
+        """
+        self.supported_versions[name.lower()] = supported_versions
+
+    def _get_run(self, run_id: str, should_exists: Optional[bool] = None) -> ToilWorkflow:
+        """
+        Helper method to instantiate a ToilWorkflow object.
+
+        :param run_id: The run ID.
+        :param should_exists: If set, ensures that the workflow run exists (or
+                              does not exist) according to the value.
+        """
+        run = ToilWorkflow(run_id, work_dir=os.path.join(self.work_dir, run_id))
+
+        if should_exists is not None:
+            if should_exists and not run.exists():
+                raise WorkflowNotFoundError
+            if should_exists is False and run.exists():
+                raise RuntimeError(f"Workflow {run_id} exists when it shouldn't.")
```

Comment:
I'd suggest creating custom Errors to replace most of our Runtime Errors.  Something like "WorkflowConflict" and a 409 added to the handle_Errors decorator.